### PR TITLE
REFTC-100 - See the Reference modifications inline (LEAN-2723)

### PR DESCRIPTION
**Pull request by asouqi**
This PR will mimic marks of([tracked_insert](https://github.com/Atypon-OpenSource/manuscripts-transform/blob/master/src/schema/marks.ts#L120) & [tracked_delete](https://github.com/Atypon-OpenSource/manuscripts-transform/blob/master/src/schema/marks.ts#L157C14-L157C28)) for meta PM node(node that has no content we render them from attributes) like citation, bibliography_item, author, affiliation, inline_equation, equation, figure, manuscript_title...

and the reason to mimic inline marks: 
- avoid adding custom logic in quarterback for block marks as that will make it hard to maintain.
- there PM node we render them in a react component like(manuscript title, author, affiliation), PM marks will not have any effect on them.

based on that we added the logic of building style of marks here to use it for both PM node and react component.

**if we need to support adding tracking marks for PM mate node will need to do the following:** 
- Node we render them with react component will need to wrap it with [TrackNodeAttrs](https://github.com/Atypon-OpenSource/manuscripts-style-guide/blob/d157611285835d3c020c2ade4d3dd6b68bed8b69/src/components/TrackMetadataPopper.tsx#L34)
- PM node we render them in the editor will need to extend their support in this [plugin](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/f7387fdb904f10370b1b61f5064ec8eb18367b47/src/plugins/tracking-mark.tsx)

CC: @mvidtman @jirikaiser-atypon @nghazzawi-atypon @IslamJMomani

### CODE REVIEW

File: src/components/TrackMetadataPopper.tsx [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+  .attrs-popper-button,
+  .attrs-popper-button-active {
+    position: absolute;
+    bottom: 0;
+    right: 0;
+  }
+`
+
+export const getMarkStyle = (dataTracked?: TrackedAttrs) => {
```
Comment (mnatsheh) [inline src/components/TrackMetadataPopper.tsx:91]: Is it possible to have this return a class name, and have the actual styles defined in `Mark`?

Comment (asouqi) [reply src/components/TrackMetadataPopper.tsx:84]: updated to return class name, but will keep the actual style so we could use it for both react component & prosemirror view

**Comment by mbartenev-atypon** [comment]
@asouqi this look ok to me.

However, I'd like to point out, as a general information for the team, that it won't be possible to use the react component for wrapping editable PM nodes in the PM view, because PM nodes will not work if wrapped into react components. While we do use react here and there in the body-editor, it's never used for the nodes, but, rather, only for the parts that are not part of the PM dom tree (popups, dropdowns etc.). This is because react rendering technics breaks selection and cursor tracking that prosemirror needs to work properly. 
cc @mvidtman @jirikaiser-atypon @IslamJMomani @nghazzawi-atypon

File: src/components/TrackMetadataPopper.tsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+  max-height: ${(props) => props.theme.grid.unit * 124}px;
+  background: ${(props) => props.theme.colors.background.secondary};
+  overflow-y: scroll;
+  overflow-x: hidden;
+
+  ::-webkit-scrollbar {
+    width: 12px;
+  }
+
+  ::-webkit-scrollbar-track {
```
Comment (mbartenev-atypon) [inline src/components/TrackMetadataPopper.tsx:308]: I would refrain from styling scrollbars both not to solve potential appearance discrepancies in different browsers on different OS and as a general accessibility practice. Also, I don't think we style scrollbars in the app in other places.

**Review by mbartenev-atypon — APPROVED** [review]

File: src/components/TrackMetadataPopper.tsx (hunk 2 above)
Comment (asouqi) [reply src/components/TrackMetadataPopper.tsx:308]: removed

**Comment by mnatsheh** [comment]
This might be a naive question, but since we're using this in both react and non-react contexts, wouldn't it be better to implement this in plain HTML? The design is quite simple, and state is really needed as far as I can tell.

**Comment by asouqi** [comment]
> wouldn't it be better to implement this in plain HTML?

that will complicate things when we need to use html element with the react component, the simplest element that we can add as plain HTML and reuse in both will be the tracking button which will injected to react using [dangerouslySetInnerHTML](https://react.dev/reference/react-dom/components/common#dangerously-setting-the-inner-html) if we need to do it for other things like mark and popup view will need a lot of vanilla js code to handle interaction, keeping them as react component will help for maintainability and readability of code

**Comment by mnatsheh** [comment]
Let's me ask a different question, where in React are we expecting to use this? shouldn't all tracked data be part of prosemirror and body-editor, and thus a plain HTML solution should be mostly sufficient? 

If we do need a React implementation, we should also consider writing 2 implementations. I understand that it's code duplication, but that might be better than the clunkiness (in my opinion) of the current solution and we already do that for figure dropdowns for example.

My concern here is that we're adding things to the body-editor interface (EditorProps) that don't belong there. Why should the article editor be responsible for implementing "addAttrsTrackingButton".

**Comment by asouqi** [comment]
> where in React are we expecting to use this? shouldn't all tracked data be part of prosemirror and body-editor, and thus a plain HTML solution should be mostly sufficient?

will use it in [author list](https://github.com/Atypon-OpenSource/manuscripts-style-guide/blob/master/src/components/AuthorsList/Author.tsx#L67C14-L67C20) and [affiliation list](https://github.com/Atypon-OpenSource/manuscripts-style-guide/blob/master/src/components/AffiliationsList.tsx#L43)

>My concern here is that we're adding things to the body-editor interface (EditorProps) that don't belong there. Why should the article editor be responsible for implementing "addAttrsTrackingButton".

updated to create a plain HTML button

**Comment by mnatsheh** [comment]
As discussed with Mikhail, we would like to have authors and affiliations moved out of react and into the body-editor (this would happen when we work on TC display for authors and affiliations), so we do not need to support a react version of the marks.

I suggest we have all the work done in body-editor, since no react is needed.

**Pull request closed without merging**